pkg/server/sandbox_status.go

```diff
@@ -66,26 +66,38 @@ func (c *criService) PodSandboxStatus(ctx context.Context, r *runtime.PodSandbox
 	}, nil
 }
 
-func (c *criService) getIP(sandbox sandboxstore.Sandbox) (string, error) {
+func (c *criService) getNetworkStatus(sandbox sandboxstore.Sandbox) (*runtime.PodSandboxNetworkStatus, error) {
 	config := sandbox.Config
+	ns := runtime.PodSandboxNetworkStatus{}
 
 	if config.GetLinux().GetSecurityContext().GetNamespaceOptions().GetNetwork() == runtime.NamespaceMode_NODE {
 		// For sandboxes using the node network we are not
 		// responsible for reporting the IP.
-		return "", nil
+		return &ns, nil
 	}
 
 	if closed, err := sandbox.NetNS.Closed(); err != nil {
-		return "", errors.Wrap(err, "check network namespace closed")
+		return nil, errors.Wrap(err, "check network namespace closed")
 	} else if closed {
-		return "", nil
+		return &ns, nil
 	}
 
-	return sandbox.IP, nil
+	ns.Ip = sandbox.IP
+	// add additional ips assigned from CNI
+	if configs, ok := sandbox.CNIResult.Interfaces[defaultIfName]; ok && len(configs.IPConfigs) > 0 {
```

Comment:
do we have any error checking that `defaultIfName < len(sandbox.CNIResult.Interfaces)` somewhere?

Reply:
the code path [here](https://github.com/kevpar/cri/blob/windows_port/pkg/server/sandbox_run.go#L82) errors out if the interfaces map doesn't have the interface name, which crashes pod creation